spice/spice.py

```diff
@@ -95,6 +97,37 @@ def result(self) -> T:
         return self._result
 
 
+class Behavior(Enum):
+    RETRY = "retry"
+    RETURN = "return"
+
+class RetryStrategy(ABC):
```

Comment:
Can you write a doc string about how this should be used and move it to its own file? And define the default validator in a separate file as well? Also define a validator that adds the model response to messages along with a message explaining what went wrong. Including the exception of there is one.